scripts/branch-deployment.sh

```diff
@@ -5,22 +5,36 @@
 #
 # External environment variables: $GITHUB_TOKEN, $PR_SOURCE_BRANCH_NAME, $SOURCE_BRANCH_NAME
 
+# Check that the needed evnironment variables are present
+if [[ -z "$SOURCE_BRANCH_NAME" && (( -z "$PR_SOURCE_BRANCH_NAME" || $PR_SOURCE_BRANCH_NAME == *"$"* )) ]]; then
+  echo "Env variable 'SOURCE_BRANCH_NAME' or 'PR_SOURCE_BRANCH_NAME' was not correctly provided. Only one env variable is necessary."
+  exit 1
+elif [[ -z "$GITHUB_TOKEN" ]]; then
+  echo "Env variable 'GITHUB_TOKEN' was not provided."
+  exit 1
+fi
+
 # If the PR_SOURCE_BRANCH_NAME contains a $ char, the variable could not be resolved, which means,
 # that the build does not originate form a pull request. In that case, we go after the branch name.
 # This is an issue with azure, that the source_branch contains whatever reference, but the actual
 # source branch that we would be interested in.
-[[ $PR_SOURCE_BRANCH_NAME != *"$"* ]] && BRANCH_NAME="$PR_SOURCE_BRANCH_NAME" || BRANCH_NAME="$SOURCE_BRANCH_NAME"
+[[ -n "$PR_SOURCE_BRANCH_NAME" && $PR_SOURCE_BRANCH_NAME != *"$"* ]] && BRANCH_NAME="$PR_SOURCE_BRANCH_NAME" || BRANCH_NAME="$SOURCE_BRANCH_NAME"
 echo "PR-Source-Branch-Name:              $PR_SOURCE_BRANCH_NAME"
 echo "Source-Branch-Name:                 $SOURCE_BRANCH_NAME"
 echo "Branch name that will be deployed:  $BRANCH_NAME"
 
+rm -rf ./dist
+
 mkdir ./dist
 git clone https://$GITHUB_TOKEN@github.com/axa-ch/plib-feature.git --depth 1 ./dist
 
 rm -rf ./dist/$BRANCH_NAME/
 mkdir -p ./dist/$BRANCH_NAME
 mv -v ./storybook-static/* ./dist/$BRANCH_NAME/
 
+# Done! From here, everything would be ready to be committed and pushed. But
+# outdated branches need to be cleaned up too, so...
```

Comment:
so.... what?! Nothing happens after that line...?!

Reply:
This is like a "where are we in the workflow". If you think that does not help, I could remove it. I just want to express somewhere what is happening at which position, but I agree that this might be confusing.

Reply:
You can rephrase it for a better understanding.